warehouse/accounts/services.py

```diff
@@ -455,6 +469,81 @@ def record_event(self, user_id, *, tag, ip_address, additional=None):
         user = self.get_user(user_id)
         return user.record_event(tag=tag, ip_address=ip_address, additional=additional)
 
+    def generate_recovery_codes(self, user_id):
+        user = self.get_user(user_id)
+
+        if user.has_recovery_codes:
+            self.db.query(RecoveryCode).filter_by(user=user).delete()
+
+        recovery_codes = [secrets.token_hex(8) for _ in range(8)]
```

Comment:
Ultra nit: Might be worth parametrizing the `8` in `range(8)` as `RECOVERY_CODE_COUNT` or similar.